test/engine.test.js

```diff
@@ -496,4 +499,133 @@ describe("Pipeline Engine tests", function () {
         const result = await pipeline.run(plan);
         assert.ok(!result.renditionErrors, `Unexepected error: ${result.renditionErrors}`);
     });
+
+    it("Should generate a presigned url when sourceType is 'URL' with input datauri", async function () {
+        mockRequire('../lib/temporary-cloud-storage', {TemporaryCloudStorage});
+        mockRequire.reRequire('../lib/sdk/storage/datauri');
+        mockRequire.reRequire('../lib/sdk/storage');
+        const Engine = mockRequire.reRequire('../lib/engine');
+
+        const pipeline = new Engine();
+        let transformerRan = false;
+        let preparedInputAsset;
+        class TestTransformer extends Transformer {
+            async compute(input) {
+                debug('Running the TestTransformer!');
+                transformerRan = true;
+                preparedInputAsset = input;
+            }
+        }
+        pipeline.registerTransformer(new TestTransformer('test'));
+
+        const plan = new Plan();
+        plan.add("test", {
+            input: {
+                type: 'image/png',
+                url: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQYV2NgYAAAAAMAAWgmWQ0AAAAASUVORK5CYII=',
+                sourceType: 'URL'
+            },
+            output: {
+                type: "image/jpeg"
+            }
+        });
+
+        await pipeline.run(plan);
+        assert.ok(transformerRan);
+        assert.ok(preparedInputAsset.url.includes('preSignUrl'));
+    });
+
+    it("Should generate a presigned url when sourceType is 'URL' without input url", async function () {
+        mockRequire('../lib/temporary-cloud-storage', {TemporaryCloudStorage});
+        const Engine = mockRequire.reRequire('../lib/engine');
+        const pipeline = new Engine();
+
+        let transformerRan = false;
+        let preparedInputAsset;
+        class TestTransformer extends Transformer {
+            async compute(input) {
+                debug('Running the TestTransformer!');
+                transformerRan = true;
+                preparedInputAsset = input;
+            }
+        }
+        pipeline.registerTransformer(new TestTransformer('test'));
+
+        const plan = new Plan();
+        plan.add("test", {
+            input: {
+                type: 'image/png',
+                path: 'fakeSuccessFilePath',
```

Comment:
what happens in `input.path` is not a valid file path?
(please add some negative tests)